src/approval/approval.service.ts

```diff
@@ -0,0 +1,96 @@
+import Axios, { AxiosResponse, AxiosInstance } from 'axios';
+import { IApproverInfo, ICanApproveToUser, IApprovalRequest } from './approvers.interface';
+import { ApprovalError, NotFoundError, ApplicationError, UnauthorizedError } from '../utils/errors/errors';
+import { config } from '../config';
+import { getToken } from "../spike/spike.service";
+import { TransferRepository } from '../transfer/transfer.repository';
+
+export class ApprovalService {
+
+    private instance: AxiosInstance;
+
+    constructor() {
+        this.instance = Axios.create({ baseURL: config.approval.approvalUrl });
+        this.addAuthIntreceptor();
+    }
+
+    async createRequest(data: IApprovalRequest) {
+        try {
+            const res: AxiosResponse = await this.instance.post(`/api/v1/request`, data);
+            return res.data;
+        } catch (err) {
+            await TransferRepository.deleteByID(data.id || "");
+            if (!err.response || !err.response.status) throw new ApplicationError();
+
+            const status: number = err.response.status;
+            if (status === 401) {
+                throw new UnauthorizedError(`request was not authorized: ${err.message}`)
```

Comment:
I wouldn't throw an `UNAUTHENTICATED` error here to the client. He doesn't even have to know that some operations in this service require authentication with other services.

Reply:
Thats right.

Reply:
Did you change the status code check from `401` to `404` because of it?